### DISCUSSION TITLE
enforcing pagination in relational resolvers

### DISCUSSION THREAD

**Discussion by shawnjones253** [MichalLytek/typegraphql-prisma#440] [Q&A]
Is there any way to enforce pagination or the usage of "take" with a maximum value?

For context, we're using the generated relational resolvers currently, but are hoping to implement graphql amount limiting / pagination across the API. Something similar to [this](https://www.apollographql.com/blog/securing-your-graphql-api-from-malicious-queries#amount-limiting).

In resolvers we've written ourselves, I'd simply replace the type of `take` in any `args` with something like `createIntScalar` from [here](https://github.com/joonhocho/graphql-scalar).

I'm hoping (though it doesn't look possible) there is a way to do this without completely abandoning the auto-generated relational resolvers.

Thanks!

**Comment by shawnjones253** [comment]
@MichalLytek -- would you be open to a PR if I could find an elegant way to work this into the generator?

**Comment by mmmeff** [comment]
Bumping, curious if there's any blessed way to customize these relational resolvers at all a la middleware or something.

If we were to take a stab at a PR for this, what do you think is the best place to hook into @MichalLytek ?

I'm torn between pursuing a configuration option in `typegraphql-prisma` vs a middleware/decorator model in `typegraphql` proper.

**Reply by mmmeff** [reply]
For context, the relational resolvers in our case can expose tables with 100,000s of records. Without enforcing pagination, this is a very easy DOS vector for either intentional or accidental exploitation.

**Reply by MichalLytek** [reply]
If you want to make a PR, please make sure to hide this behind a generator option, like with the "Emit `isAbstract` decorator option".
